File: typescript/src/chart/ChartConstants.ts

class ChartConstants {
    public get supportedFormatType(): string[] {
        return ["json"];
    }

    public get getChartDetailUrl(): string {
        return `https://api.dapplooker.com/chart`;
    }

    public get timeoutLimit(): number{
        return 15000; // 15 sec connection timeout
    }
}

export default new ChartConstants();

File: go/dapplooker/chartConstants.go
package dapplooker

import "time"

type ChartConstants struct{}

func (c *ChartConstants) SupportedFormatType() []string {
	return []string{"json"}
}

func (c *ChartConstants) GetChartDetailUrl() string {
	return "https://api.dapplooker.com/chart"
}

func (c *ChartConstants) TimeoutLimit() time.Duration {
	return 15 * time.Second
}
